### PR TITLE
Calculate positional indices of operands to operations.

### PR DESCRIPTION
**Pull request by markww**
Because lists in Souffle are just recursive linked-list structs, it is
not very natural to look up elements by index. For the tag transform
operation from the SQL verifier, however, we need to do exactly that.
This commit introduces `operationHasOperandAtIndex`, which enriches the
old `operationToOperands` relation by adding the position the operand
holds in the operation's `operandList`.

### CODE REVIEW

File: src/analysis/souffle/operations.dl [hunk 1]
```diff
@@ -48,11 +48,32 @@ operationToPartialOperandLists([owner, op, result, operandList, attrs], operandL
 operationToPartialOperandLists(operation, tail) :-
   operationToPartialOperandLists(operation, [_, tail]).
 
-// Map full operations to operands.
-.decl operationToOperands(operation:Operation, operand: AccessPath)
+// `listLength` is the length of the `operandList`.
+.decl operandListLength(operandList: OperandList, listLength: number)
```
Comment (bgogul) [inline src/analysis/souffle/operations.dl:52]: Should I just drop #548 and you can fix the names in this PR? Or, should I try to merge it in?

File: src/analysis/souffle/operations.dl [hunk 2]
```diff
@@ -48,11 +48,32 @@ operationToPartialOperandLists([owner, op, result, operandList, attrs], operandL
...
 
-operationToOperands(operation, operand) :-
-  isOperation(operation), operationToPartialOperandLists(operation, [operand, _]).
+operandListLength(nil, 0).
+operandListLength([head, tail], tailLen + 1) :-
+  operationToPartialOperandLists(_, [head, tail]), operandListLength(tail, tailLen).
```
Comment (bgogul) [inline src/analysis/souffle/operations.dl:56]: If you decide to do the renaming in #548, please change this to `operationHasOperandListSuffix`. Similarly, for `operationToPartialAttributeLists`

**Review by bgogul — APPROVED** [review]

File: src/analysis/souffle/operations.dl (hunk 1 above)
Comment (markww) [reply src/analysis/souffle/operations.dl:52]: Maybe that would be best. If we both submitted, there would be some significant conflicts that would just duplicate the work anyway.

Comment (bgogul) [reply src/analysis/souffle/operations.dl:52]: Sounds good. I will close the other PR.

File: src/analysis/souffle/operations.dl (hunk 2 above)
Comment (markww) [reply src/analysis/souffle/operations.dl]: Done.

**Pull request closed without merging**